### PR TITLE
[scala] remove long-deprecated Ensime integration

### PR DESCRIPTION
**Pull request by Kazark**
According to its website, the Ensime project has been shut down since 2019. We have been supporting both Ensime and Metals for [a year and a half](https://github.com/syl20bnr/spacemacs/pull/12234#issuecomment-524916394). Ensime's GitHub repositories, including its Emacs integration, are archived. Meanwhile, Metals has been developed actively, regularly releasing new versions. It has stabilized significantly, and supports an increasingly full set of features. Scala 3 is just around the corner, and the community is poised to make the transition smoothly and relatively quickly. Metals supports it already, whereas of course Ensime does not and never will. In fact, Scala 2 has had several import minor versions released since Ensime died. Now that it's 2021, it's time to cut the baggage we are carrying around for Ensime.

### CODE REVIEW

File: layers/+lang/scala/layers.el [hunk 1]
```diff
@@ -9,7 +9,7 @@
 ;;
 ;;; License: GPLv3
 
-(configuration-layer/declare-layer-dependencies '(java))
+(configuration-layer/declare-layer-dependencies '(java lsp))
```
Comment (smile13241324) [inline layers/+lang/scala/layers.el:12]: This is not necessary, `dap` layer already includes lsp

File: layers/+lang/scala/config.el [hunk 2]
```diff
@@ -11,24 +11,21 @@
...
+(defconst scala-backends '(scala-metals)
   "Backend server implementation to enable advanced IDE language features")
 
-(defvar scala-backend 'scala-ensime
+(defvar scala-backend 'scala-metals
   "Backend used to trigger IDE language features.
-`scala-ensime' or `scala-metals' are currently supported")
+Only `scala-metals' is currently supported.")
 
 (defvar scala-auto-start-backend nil
```
Comment (smile13241324) [inline layers/+lang/scala/config.el:27]: This variable must be removed too, it was only used by ensime, lsp servers are always auto-started.

**Review by smile13241324 — CHANGES_REQUESTED** [review]
I like cutting this old piece of code however I have found some smaller issues which must be fixed before we can merge this.

* Some variables are now obsolete and should be removed also
* As far as I see there is no REPL support available anymore, I don't think metals has one, so the feature must be removed
* Have you made a test on your local config that metals still correctly loads?

File: layers/+lang/scala/config.el (hunk 2 above)
Comment (Kazark) [reply layers/+lang/scala/config.el:27]: That's why I love deleting. I always learn something. 😁

File: layers/+lang/scala/layers.el (hunk 1 above)
Comment (Kazark) [reply layers/+lang/scala/layers.el:12]: It was necessary for me. I did not get either `dap` or `lsp` keybindings in my major mode map for Scala until I added this.

**Comment by Kazark** [comment]
>  Have you made a test on your local config that Metals still correctly loads?

Yes, and Scala is what I write at work so I will notice fast if something is wrong.

**Comment by Kazark** [comment]
@smile13241324 thanks for your corrections. I have rebased and made the revisions you requested, with the exception of removing `lsp` as a dependency, which was necessary to get this working for me. Let me know if I still missed something. 😁

**Comment by Kazark** [comment]
Also, I have retested and my Scala environment is still working correctly.

**Comment by smile13241324** [comment]
I will have a look

File: layers/+lang/scala/layers.el (hunk 1 above)
Comment (smile13241324) [reply layers/+lang/scala/layers.el:12]: Strange, anyway requiring it here too does not do any harm, so lets keep it.

**Review by smile13241324 — APPROVED** [review]

**Comment by smile13241324** [comment]
Now its cool, lets merge it :)

**Comment by hxx0215** [comment]
Since ensime seems relaunched. Would you consider to recover it?
[ensime-the-next-generation](https://contributors.scala-lang.org/t/ensime-the-next-generation/5474)

**Pull request merged**